src/corehost/cli/libhost.cpp

```diff
@@ -51,18 +51,28 @@ void get_runtime_config_paths(const pal::string_t& path, const pal::string_t& na
 
 host_mode_t detect_operating_mode(const host_startup_info_t& host_info)
 {
-    if (coreclr_exists_in_dir(host_info.dotnet_root) || pal::file_exists(host_info.app_path))
+    bool has_local_coreclr = coreclr_exists_in_dir(host_info.dotnet_root);
+    if (has_local_coreclr || pal::file_exists(host_info.app_path))
     {
-        pal::string_t own_deps_json = host_info.dotnet_root;
-        pal::string_t own_deps_filename = host_info.get_app_name() + _X(".deps.json");
-        pal::string_t own_config_filename = host_info.get_app_name() + _X(".runtimeconfig.json");
-        append_path(&own_deps_json, own_deps_filename.c_str());
+        pal::string_t deps_in_dotnet_root = host_info.dotnet_root;
+        pal::string_t deps_filename = host_info.get_app_name() + _X(".deps.json");
+        append_path(&deps_in_dotnet_root, deps_filename.c_str());
         if (trace::is_enabled())
         {
-            trace::info(_X("Detecting mode... CoreCLR present in dotnet root [%s] and checking if [%s] file present=[%d]"),
-                host_info.dotnet_root.c_str(), own_deps_filename.c_str(), pal::file_exists(own_deps_json));
+            trace::info(_X("Detecting mode... Checking CoreCLR presence in dotnet root [%s]=[%d] and checking presence of [%s]=[%d]"),
+                host_info.dotnet_root.c_str(), has_local_coreclr, deps_filename.c_str(), pal::file_exists(deps_in_dotnet_root));
         }
-        return ((pal::file_exists(own_deps_json) || !pal::file_exists(own_config_filename)) && pal::file_exists(host_info.app_path)) ? host_mode_t::apphost : host_mode_t::split_fx;
+
+        if (has_local_coreclr)
+        {
+            // Name of runtimeconfig file; since no path it is checked for in current working directory
+            pal::string_t config_in_cwd = host_info.get_app_name() + _X(".runtimeconfig.json");
+
+            // Detect standalone apphost or legacy split mode (specifying --depsfile and --runtimeconfig)
+            return (pal::file_exists(deps_in_dotnet_root) || !pal::file_exists(config_in_cwd)) && pal::file_exists(host_info.app_path) ? host_mode_t::apphost : host_mode_t::split_fx;
```

Comment:
`pal::file_exists(host_info.app_path)` is checked twice.  Can that be moved to a local variable as well?

Reply:
I refactored the code to make it easier to understand; the check is only done once now. Also, the trace message is only included in the standalone\split check now since it didn't really add anything otherwise

Reply:
I like the new refactoring.  thanks!